plc4j/api/src/main/java/org/apache/plc4x/java/api/messages/PlcReadRequest.java

```diff
@@ -38,6 +39,16 @@ interface Builder extends PlcRequestBuilder {
         Builder addTagAddress(String name, String tagAddress);
         Builder addTag(String name, PlcTag tag);
 
+        /**
+         * Parses a string representation of a tag address into a {@link PlcTag} object.
+         * <p>
+         * This method safely attempts to convert the given tag address string into a PlcTag, returning an
+         * {@link Optional} that is empty if the parsing fails.
+         *
+         * @param tagAddress The string representation of the tag address to parse.
+         * @return An optional holding the parsed PLC tag if successful, otherwise an empty optional.
+         */
+        Optional<PlcTag> parseSafe(String tagAddress);
```

Comment:
I would prefer "parseTagAddress" for this to match the naming pattern of the other methods.

Reply:
Also not 100% sure, if we should not instead have this method in the connection instead ... reasoning is that when adding it to the builder it doesn't really add anything to the request it builds .... however having a "parseTagAddress" and "parseQueryAddress" in the connection sounds like a very cool thing to do, as this way tooling utilizing PLC4X can use to validate tags and queries.